### PR TITLE
STN-324 : Detail Page Image Styles

### PR DESCRIPTION
**Pull request by heatherdesigns**
# [STN-324](https://sparkbox.atlassian.net/browse/STN-324) READY FOR REVIEW

## Summary
The work in this PR empowers H&S to display images using different images styles when the need arises.

### Acceptance Criteria
* Identify a solution for displaying a non-resized featured-image, within the 3-column-with-image layout, that meets H&S’s needs.
* Ensure that the solution allows for selection of different image styles.
* Implement the approach and demonstrate it’s functionality on the publications detail pages.
* Ensure the approach meets our browser support and accessibility guidelines.

## Need Review By (Date)
4/27

## Urgency
medium

## Steps to Test
1. In the CLI, run `npm run test` and confirm all of our Sass tests pass.
2. Run `npm start` to build the styles. The compiled CSS files were added during a rebase. 🤷  I will rebuild and update them once everything has been reviewed and approved.
2. Review the following pages with the default image style of `Medium Square 360 x 360`
    * You can **check / update the image style** used on a detail page by going to `Structure > Content types > Event _(or whichever node detail page you are working on)_ > Manage Display`. You can also use the contextual link located on the detail page.
    * Click on `Manage layout`
    * Find the `"Image" field` block, click on the contextual link, and select `Configure` 
    * Find `Image Style` and confirm / select `Medium Square (360 x 360)`
    * Update. Save layout. Back to site.
- [ ] Event: http://economics.suhumsci.loc/events/events-all-fields-filled
- [ ] Publication: http://economics.suhumsci.loc/publications/publications-all-fields-filled
- [ ] News: http://economics.suhumsci.loc/news/joeys-test-news-item
- [ ] Person: http://economics.suhumsci.loc/people/ran-abramitzky
3. The work in this PR gives H&S the flexibility to use the image styles set in the configuration. I will be adding recommendations on which image styles to use to the Pattern Doc. Following the steps above, test various image styles: `Medium Scaled (360)`, `Small Square (200 x 200)`, etc.
4. Test another option. 
H&S requested styles for when the image is placed within the main body.
    * Go to http://economics.suhumsci.loc/publications/publications-all-fields-filled
    * Click on the contextual link and select `Manage Display`
    * Click on `Manage layout`
    * Move `"Publications Image"` into the main body section
    * Click on the contextual link for the publications image
    * Add the utility class of `hb-main-body-detail-image` to  the `Field Formatter Class`
    * Update. Save layout. Back to site.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)

### CODE REVIEW

File: docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.three_column_w_image.scss [hunk 1]
```diff
@@ -51,45 +51,32 @@
   &__main-top {
     display: flex;
     flex-wrap: wrap;
-    align-items: flex-start;
+    justify-content: space-between;
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.three_column_w_image.scss:54]: 👍 much better

File: docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.three_column_w_image.scss [hunk 2]
```diff
@@ -51,45 +51,32 @@
...
       @include grid-media-min('md') {
-        width: 60%;
         display: flex;
         flex-direction: column;
       }
     }
   }
 
   &__main-top-image {
-    width: 100%;
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.three_column_w_image.scss:91]: hmm I think at mobile, this is ok?

File: docroot/themes/humsci/humsci_basic/src/scss/utilities/_general.scss [hunk 3]
```diff
@@ -87,3 +87,15 @@
     }
   }
 }
+
+// Used specifically on node detail pages when the image
+// is placed within the main content of the page in layout builder
+.hb-main-body-detail-image {
+  .hb-three-column-w-image & {
+    @include grid-media-min('md') {
+      @include clearfix;
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/utilities/_general.scss:96]: 👍 looks good!

**Review by silverli — CHANGES_REQUESTED** [review]
- Tests are passing 👍 
- All the image styles are looking good both in the top and in the main section 👍 (except I have to do a cache clear 
- I checked some of the layouts on the node detail pages without a sidebar, and with two sidebars as well, everything looks good 👍 

every time I make a change which is interesting)

Two questions:
- I asked about keeping the img width 100% on mobile -- we can let height do it's thing, but I do think it's nice on mobile

- Also, did you do browser testing?

File: docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.three_column_w_image.scss (hunk 2 above)
Comment (heatherdesigns) [reply docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.three_column_w_image.scss:91]: This property doesn't do anything now that we are using the Drupal image styles.

**Comment by heatherdesigns** [comment]
@silverli Yes! Browser testing has been completed. 😄

File: docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.three_column_w_image.scss (hunk 2 above)
Comment (heatherdesigns) [reply docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.three_column_w_image.scss:91]: A note on the functionality of images now that we are using the Drupal image style functionality to select different image styles (sizes and dimensions): Only images medium or larger scale to 100% on mobile. Once the image is larger than the mobile screen size, they scale down to fit. Small images are only 200px width therefore will not scale at mobile. Adding styles to make small image style will cause the image to pixelate which is not recommended.

Image will also wrap and scale at different breakpoints depending upon the length of the text in the title.

Comment (heatherdesigns) [reply docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.three_column_w_image.scss:91]: I will add this to the code base.

File: docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.three_column_w_image.scss [hunk 4]
```diff
@@ -1,3 +1,11 @@
+// An note on the functionality of images
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.three_column_w_image.scss]: ah sorry. should this content go on line 87 or 112?

File: docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.three_column_w_image.scss [hunk 5]
```diff
@@ -1,3 +1,11 @@
...
+// --------------------------------------
+// The height and width of images are set and can be changed in layout builder iamge styles.
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.three_column_w_image.scss]: iamge -> image

Comment (silverli) [reply docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.three_column_w_image.scss]: and maybe also say instead: layout build image styles -> layout builder configuration for images

**Review by silverli — CHANGES_REQUESTED** [review]
gah sorry one more little thing

File: docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.three_column_w_image.scss (hunk 5 above)
Comment (heatherdesigns) [reply docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.three_column_w_image.scss]: I like that!! 👍

File: docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.three_column_w_image.scss (hunk 4 above)
Comment (heatherdesigns) [reply docroot/themes/humsci/humsci_basic/src/scss/objects/_layouts.three_column_w_image.scss]: I like it on line 87. 😄

**Pull request merged**